mlir/lib/Dialect/MIOpenOps/Driver/mlir-miopen-driver.cpp

```diff
@@ -220,29 +231,85 @@ int main(int argc, char **argv) {
-      builder.getNamedAttr("output_layout", builder.getArrayAttr(ArrayRef<Attribute>(outputLayoutSpec.begin(), outputLayoutSpec.end()))),
-
-      builder.getNamedAttr("dilations", builder.getArrayAttr({
-                                          builder.getI32IntegerAttr(dilationHeight.getValue()),
-                                          builder.getI32IntegerAttr(dilationWidth.getValue()),
-                                        })),
-      builder.getNamedAttr("strides", builder.getArrayAttr({
-                                          builder.getI32IntegerAttr(strideHeight.getValue()),
-                                          builder.getI32IntegerAttr(strideWidth.getValue()),
-                                        })),
-      builder.getNamedAttr("padding", builder.getArrayAttr({
-                                          builder.getI32IntegerAttr(paddingHeight.getValue()),
-                                          builder.getI32IntegerAttr(paddingWidth.getValue()),
-                                        })),
-    });
-  block->push_back(convOp);
+  if (convFlavor.getValue().compare("conv2d") == 0) {
+    auto convOp = builder.create<miopen::Conv2DOp>(
+        builder.getUnknownLoc(), ArrayRef<Type>({}),
+        ValueRange{block->getArgument(0), block->getArgument(1),
+                   block->getArgument(2)},
+        ArrayRef<NamedAttribute>{
+            builder.getNamedAttr(
+                "filter_layout",
+                builder.getArrayAttr(ArrayRef<Attribute>(
+                    filterLayoutSpec.begin(), filterLayoutSpec.end()))),
+            builder.getNamedAttr(
+                "input_layout",
+                builder.getArrayAttr(ArrayRef<Attribute>(
+                    inputLayoutSpec.begin(), inputLayoutSpec.end()))),
+            builder.getNamedAttr(
+                "output_layout",
+                builder.getArrayAttr(ArrayRef<Attribute>(
+                    outputLayoutSpec.begin(), outputLayoutSpec.end()))),
+
+            builder.getNamedAttr(
+                "dilations",
+                builder.getArrayAttr({
+                    builder.getI32IntegerAttr(dilationHeight.getValue()),
+                    builder.getI32IntegerAttr(dilationWidth.getValue()),
+                })),
+            builder.getNamedAttr(
+                "strides",
+                builder.getArrayAttr({
+                    builder.getI32IntegerAttr(strideHeight.getValue()),
+                    builder.getI32IntegerAttr(strideWidth.getValue()),
+                })),
+            builder.getNamedAttr(
+                "padding",
+                builder.getArrayAttr({
+                    builder.getI32IntegerAttr(paddingHeight.getValue()),
+                    builder.getI32IntegerAttr(paddingWidth.getValue()),
+                })),
+        });
+    block->push_back(convOp);
+  } else if (convFlavor.getValue().compare("conv2d_bwd_data") == 0) {
```

Comment:
codes here looks duplicated, could attributes be assigned in a lambda function?

Reply:
Sounds like a good idea.